GANDLF/losses/loss_calculators.py

```diff
@@ -0,0 +1,106 @@
+import torch
+import torch.nn.functional as F
+from GANDLF.losses import get_loss
+from GANDLF.utils.tensor import reverse_one_hot, get_linear_interpolation_mode
+from abc import ABC, abstractmethod
+from typing import List
+
+
+class AbstractLossCalculator(ABC):
```

Comment:
I did, and I also agree with Slava. Packing the losses into classes that implement the __call__ method would basically let us keep the existing codebase nearly intact while giving more organization to the code. Should I do that in the meantime? I can open a separate PR.

And regarding this abstract interface - I think we will still need it, as it addresses some specific combinations and operations that are related to the model, so using the losses but in a specific way. What do you think?

Reply:
Btw regarding those - what do you think, if we decide to implement the common interface for losses, that the params are an instance variable set in the initialization and the call only uses necessary inputs? We could hide this initialization behind the get_loss method, so once again from perspective of codebase it will be invisible

Reply:
Completely agree. 👍🏽

Reply:
noted, will create pr for that